packages/plugin-ext/src/plugin/plugin-context.ts

```diff
@@ -215,14 +215,40 @@ export function createAPIFactory(
 
     return function (plugin: InternalPlugin): typeof theia {
         const authentication: typeof theia.authentication = {
-            registerAuthenticationProvider(provider: theia.AuthenticationProvider): theia.Disposable {
-                return authenticationExt.registerAuthenticationProvider(provider);
+            // support older (< 1.53.0) and newer version of authentication provider registration
+            registerAuthenticationProvider(
+                id: string | theia.AuthenticationProvider, label?: string, provider?: theia.AuthenticationProvider, options?: theia.AuthenticationProviderOptions):
+                theia.Disposable {
```

Comment:
Suggestion: This implementation seems a bit convoluted to me. Perhaps we could use something like this?

```ts
registerAuthenticationProvider(id: string | theia.AuthenticationProvider, label?: string, provider?: theia.AuthenticationProvider, options?: theia.AuthenticationProviderOptions): theia.Disposable {
    // Add some comment here about why we need the merged definition
    type MergedAuthProvider = theia.AuthenticationProvider & {
        createSession(scoped: string[]): Thenable<theia.AuthenticationSession>
        removeSession(sessionId: string): Thenable<void>
        getSessionsWithoutScopes(): Thenable<ReadonlyArray<theia.AuthenticationSession>>
    };
    const isNew = typeof id === 'string';
    const providerId = isNew ? id : id.id;
    const mergedProvider = (isNew ? provider : id) as MergedAuthProvider;

    if (!mergedProvider.createSession) {
         mergedProvider.createSession = (scopes: string[]) => mergedProvider.login(scopes);
    }
    if (!mergedProvider.removeSession) {
        mergedProvider.removeSession = (sessionId: string) => mergedProvider.logout(sessionId);
    }
    if (!mergedProvider.getSessionsWithoutScopes) {
         mergedProvider.getSessionsWithoutScopes = mergedProvider.getSessions;
    }
    return authenticationExt.registerAuthenticationProvider(providerId, label!, mergedProvider, isNew ? options : {
          supportsMultipleAccounts: mergedProvider.supportsMultipleAccounts
    });
},
```

As far as I can tell, we don't need the following bit:

```ts
newProvider['getSessions'] = (_scopes?: string[]) => (newProvider as any)['getSessionsWithoutScopes']();
```

Reply:
I see, I didn't look into it too much and thought `getSessionsWithoutScopes` was part of the API. Yeah, JS runtimes don't care about parameters :)